### ISSUE TITLE
Feature Request: Start Session as Council Member

### ISSUE THREAD

**Issue by Morcsine** [evil-morfar/RCLootCouncil2#219]
We want to have Group Loot in our Raid and sort the Loot in Categories as Caster and Melee Stuff.
It would be fine if there is an Option that every Council Member(or a certain Raid Member beside the Group Leader) could start a Session.

With the Speedrun Meta and small Bags it is awful to have the Masterlooter/Group Leader have all Loot in his Bags and control the Loot himself.

**Comment by zarillion** [comment]
I would like to second this. I am the raid leader and I have one of my officers handle loot. He currently needs to have raid leader in game for the addon to function, which breaks functionality in other addons which would require me to be raid leader.

It would be nice if he could at minimum start sessions and award the loot by being on the council, so I could keep the raid leader role in game and use the other addons as intended. Ideally it would go one step further and I could pick the master looter when I start an RCLC session, and all the loot goes to him as well so I do not have to hand it out either. Everyone else's addon would trust that he is the master looter because the addon message assigning him as master looter came from the raid leader.

**Comment by evil-morfar** [comment]
There's a big difference in assigning someone else as the ML and allowing others (e.g. raid assistants) to control sessions.

Unfortunately "because the leader says he's the ML" is not trivial - but if you can come up with a deterministic way of near instantly determining whom the ML is I'd do it. Or at the very least figure out how to queue comms while determining it without breaking anything.

**Comment by Morcsine** [comment]
When i get it right the addon can´t queue commands from 2 Players in one Session and would break ?

**Comment by evil-morfar** [comment]
> When i get it right the addon can´t queue commands from 2 Players in one Session and would break ?

That's not really the issue.

Say we're in a raid, I'm a raider, and I just did a reload. Now I need to know who the ML is, if we're using the addon, and get his settings. To determine the ML I would need to ask the group leader and then I can ask the ML for the settings. But in the mean time, people are sending responses and I maybe even get something from the ML, all of which I have to discard because I need to wait for data syncing back and forth - something I right now can do instantly.

And that's just the tip of the iceberg. Every single command sent between clients could at any time rely on this extra step, not to mention how to even build an interface for selecting the ML.

**Comment by zarillion** [comment]
I agree it definitely makes it more complex to pick a separate master looter.

In the mean time allowing council/raid assistance to start sessions and award the loot winners while I as the raid leader / ML still get the loot and hand it out would be a huge improvement for us!

**Comment by Dysonfm** [comment]
I agree with this. (this thread was linked based on an idea I was posting on discord).

One additional idea to consider to make the alternative master looter more definitive -- use officer note like EPGP does. Officer notes are often not used (hence why EPGP uses it as a storage) and can be locked down so only officers write and others can see. You could coordinate that everyone in raid selects that they will scan officer note of people online very quickly and determine from there the master looter. The 'backup' master looter is always raid leader, but the primary would be the one with a codeword like <ML> contained in the officer note of a single individual. You could also have <ML> and <ML2> so if you scan the officer notes and <ML> is offline, <ML2> is the backup before going to raid lead default.

Scanning officer notes shouldnt be terribly resource intense on a re-load because EPGP addons have done this method for over a decade I think?  For the record I dont use EPGP, and I know this might conflict with an EPGP addon layered into RCLC, but for those that use it without EPGP, the officer note concept may work. Something to consider. 

Caveat: I dont know how officer notes work anymore because I am not an officer, but reviewing EPGP addons, they all claim to use officer notes as the database for EP/GP storage I think, so this is why I imagine it is possible.

**Comment by evil-morfar** [comment]
@Dysonfm and what would happen when two people are marked for it?

**Comment by Dysonfm** [comment]
A reasonable 'convention' if two officer notes have the <ML> or :ML: or whatever tag you'd want in the officer note to cue that the person should be the 'real master looter' (not raid leader) is that if there is an odd conflict where two folks have an ML tag, that the addon would just simply default loot master back to being Raid Lead because it cannot figure it out, and then print in the chat log an error note saying "more than 1 officer note designated ML" would be the 'cue' that something is wrong. Or throw up a dialog box to the RL upon re-load letting them know "2 people have ML designated in Officer Note, therefore reverting back to you".

If you want to make it possible that more than one person can be alternative-to-RL master loot, you could allow a priority choice system. A tag like "<ML>" for the primary, and if that person isn't on, it searches for someone else who might have a tag of "<ML2>" as the backup and if no "<ML2>" tag, then it reverts back to RL. After all, the person with <ML> tag in the officer note might not even be online, maybe they are sitting out for that boss because of raid comp for a mythic boss or whatnot, then there should always be a "backup" ML and the default backup would always end up being the RL.

**Comment by evil-morfar** [comment]
A huge problem is it only works with people you're in guild with though.

**Comment by nivthefox** [comment]
Why not allow the Raid Leader to have a setting in their addon which denotes who the Master Looter is?  Then, any time Master Looter functionality is needed, it checks with the RL first, then transfers to the ML if the RL has designated someone? That would be a deterministic way to do this which should be automatable, right?

**Comment by evil-morfar** [comment]
@nivthefox https://github.com/evil-morfar/RCLootCouncil2/issues/219#issuecomment-1367566706

**Comment by nivthefox** [comment]
Ah, I do see you've addressed that, but I do have a suggested fix, there, too: if the Raid  Leader doesn't want to be master looter, then require them to sync their settings with who they do want to be Master Looter.  Then, you can get settings from either RL or ML, and you will have the same settings. If at any point their settings are out of sync, throw a big error at both of them and make them resolve it.

**Comment by Raistlin1990** [comment]
similar topic and interesting to see that there is an ongoing discussion about this and technical limitations.

Say you are the ML and running the RC Session. RC can only assign loot, when the loot window is open. The player should be given an item, but he is out of instance. How do I assign it now? Should I use "award later" for this? I don't really have time in raid to figure things out, due to our RL

**Comment by evil-morfar** [comment]
> similar topic and interesting to see that there is an ongoing discussion about this and technical limitations.
> 
> Say you are the ML and running the RC Session. RC can only assign loot, when the loot window is open. The player should be given an item, but he is out of instance. How do I assign it now? Should I use "award later" for this? I don't really have time in raid to figure things out, due to our RL

Tell your raiders not to leave the instance if they want loot - that's a restriction imposed by Blizzard (you can't ML items to people not in your instance).

I'd award the item to myself, then reaward it to the intended player. This will ML the item to you, removing the restriction, and allowing you to trade the item to the winner.

**Comment by evil-morfar** [comment]
@Raistlin1990
>so its nice if the data is correct

That's why I mentioned reaward - then you won't have to edit the history, it does so automatically.

>  I ML it to me because "Can't assign items without the loot window open"

In classic (btw this is *not* the classic repo) there's several ways to do loot. The standard is ML, where you can hand out items directly to players "using" the default master loot setup provided by Blizzard. This does not work without the default loot list being open, as you must have access to the item to hand it out (including awarding to yourself)- that's the issue you're facing, and that's completely different from the "out of instances" you first stated. You *must* have the list open in order to award items.

Alternatively you can use "Award Later" - the key difference here is that all items are immediately awarded to you (using the ML method above), which frees you from staying at the boss with the loot list open - only downside is you must trade the items to the winners, and respect the trade timer on items.
